mongo/mongo_test.go

```diff
@@ -342,6 +346,34 @@ func TestMongoHelpers(t *testing.T) {
 				true,
 				nil,
 			},
+			{
+				"semantic single document/bson.D",
+				bson.D{{"x", 1}},
+				nil,
+				false,
+				errors.New("primitive.D is not an allowed pipeline type as it represents a single document. Use bson.A or mongo.Pipeline instead"),
```

Comment:
The `%T` `fmt` verb interprets `bson.D` as `primitive.D` since `bson.D` is actually just a [type-alias](https://github.com/mongodb/mongo-go-driver/blob/master/bson/bson.go#L33) of `primitive.D`.